benchmarks/graphics_pipeline/main.cpp

```diff
@@ -549,37 +646,44 @@ void ProjApp::Setup()
                     vertexDataCompressed.position = half3(glm::packHalf1x16(vertexData.position.x), glm::packHalf1x16(vertexData.position.y), glm::packHalf1x16(vertexData.position.z));
-                    vertexDataCompressed.texCoord = half2(glm::packHalf1x16(vertexData.texCoord.x), glm::packHalf1x16(vertexData.texCoord.y));
-                    vertexDataCompressed.normal   = i8vec4(MapFloatToInt8(vertexData.normal.x), MapFloatToInt8(vertexData.normal.y), MapFloatToInt8(vertexData.normal.z), MapFloatToInt8(1.0f));
-                    vertexDataCompressed.tangent  = i8vec4(MapFloatToInt8(vertexData.tangent.x), MapFloatToInt8(vertexData.tangent.y), MapFloatToInt8(vertexData.tangent.z), MapFloatToInt8(vertexData.tangent.a));
-                    lowPrecisionInterleaved.AppendVertexData(vertexDataCompressed);
-                    lowPrecisionPositionPlanar.AppendVertexData(vertexDataCompressed);
-
-                    highPrecisionInterleaved.AppendVertexData(vertexData);
-                    highPrecisionPositionPlanar.AppendVertexData(vertexData);
+
+                    OverwritePositionData(lowInterleavedPositionBuffer, vertexDataCompressed, i * sphereVertexCount + vertexIndex);
+                    OverwritePositionData(lowPlanarPositionBuffer, vertexDataCompressed, i * sphereVertexCount + vertexIndex);
+                    OverwritePositionData(highInterleavedPositionBuffer, vertexData, i * sphereVertexCount + vertexIndex);
+                    OverwritePositionData(highPlanarPositionBuffer, vertexData, i * sphereVertexCount + vertexIndex);
                 }
+
                 // Iterate the meshes triangles and add the vertex indices
                 for (uint32_t triIndex = 0; triIndex < sphereTriCount; ++triIndex) {
                     uint32_t v0 = PPX_VALUE_IGNORED;
                     uint32_t v1 = PPX_VALUE_IGNORED;
                     uint32_t v2 = PPX_VALUE_IGNORED;
                     mesh.GetTriangle(triIndex, v0, v1, v2);
+                    // The planar ones are the same, can just be copied later
                     lowPrecisionInterleaved.AppendIndicesTriangle(v0 + i * sphereVertexCount, v1 + i * sphereVertexCount, v2 + i * sphereVertexCount);
-                    lowPrecisionPositionPlanar.AppendIndicesTriangle(v0 + i * sphereVertexCount, v1 + i * sphereVertexCount, v2 + i * sphereVertexCount);
                     highPrecisionInterleaved.AppendIndicesTriangle(v0 + i * sphereVertexCount, v1 + i * sphereVertexCount, v2 + i * sphereVertexCount);
-                    highPrecisionPositionPlanar.AppendIndicesTriangle(v0 + i * sphereVertexCount, v1 + i * sphereVertexCount, v2 + i * sphereVertexCount);
                 }
             }
-            // Create a giant vertex buffer to accommodate all copies of the sphere mesh
+
+            // Overwrite the position vertex buffers in the geometries
+            lowPrecisionInterleaved.SetVertexBuffer(0, lowInterleavedPositionBuffer);
```

Comment:
Would you be able to remove the need to set these if Get*Buffer returned non-const? I think it's just SetSize that you can't call now?

Reply:
Yes, I chose to use setters and make copies of the buffers rather than changing the getters to be non-const, but I think that can be an option too. Is that preferred?

Reply:
I think it's better to convert Get*Buffer to be non-const instead of creating copies of buffer and overriding them, it sounds like it should be a valid usage to manipulate data in the buffers, so there should geometry should allow doing that. Would that work for your use case?